File: backend/ŠišAppApi/Models/Review.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ŠišAppApi.Models;

public class Review
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    public int UserId { get; set; }
    
    [ForeignKey("UserId")]
    public User User { get; set; } = null!;
    
    [Required]
    public int BarberId { get; set; }
    
    [ForeignKey("BarberId")]
    public Barber Barber { get; set; } = null!;
    
    [Required]
    public int AppointmentId { get; set; }
    
    [ForeignKey("AppointmentId")]
    public Appointment Appointment { get; set; } = null!;
    
    public int? SalonId { get; set; }
    
    [ForeignKey("SalonId")]
    public Salon? Salon { get; set; }
    
    public int? CustomerId { get; set; }
    
    [ForeignKey("CustomerId")]
    public Customer? Customer { get; set; }
    
    [Required]
    [Range(1, 5)]
    public int Rating { get; set; }
    
    public string? Comment { get; set; }
    
    public string? ImageIds { get; set; } // JSON array of image IDs
    
    public bool IsVerified { get; set; } = false;
    
    public bool IsHidden { get; set; } = false;
    
    public int HelpfulCount { get; set; } = 0;
    
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime? UpdatedAt { get; set; }
    
    public bool IsDeleted { get; set; } = false;
    
    public DateTime? DeletedAt { get; set; }
} 

File: sisapp_frontend/lib/models/review.dart
class Review {
  final int? id;
  final int userId;
  final String userName;
  final int barberId;
  final String barberName;
  final int? salonId;
  final String? salonName;
  final int appointmentId;
  final String? serviceName;
  final int rating;
  final String comment;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final int helpfulCount;
  final bool isVerified;
  final String? barberResponse;
  final DateTime? barberRespondedAt;

  Review({
    this.id,
    required this.userId,
    required this.userName,
    required this.barberId,
    required this.barberName,
    this.salonId,
    this.salonName,
    required this.appointmentId,
    this.serviceName,
    required this.rating,
    required this.comment,
    required this.createdAt,
    this.updatedAt,
    this.helpfulCount = 0,
    this.isVerified = false,
    this.barberResponse,
    this.barberRespondedAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      userId: (json['userId'] as num?)?.toInt() ?? 0,
      userName: json['userName'] ?? 'Anoniman',
      barberId: (json['barberId'] as num?)?.toInt() ?? 0,
      barberName: json['barberName'] ?? 'Nepoznato',
      salonId: (json['salonId'] as num?)?.toInt(),
      salonName: json['salonName'],
      appointmentId: (json['appointmentId'] as num?)?.toInt() ?? 0,
      serviceName: json['serviceName'],
      rating: (json['rating'] as num?)?.toInt() ?? 0,
      comment: json['comment'] ?? '',
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
      helpfulCount: (json['helpfulCount'] as num?)?.toInt() ?? 0,
      isVerified: json['isVerified'] ?? false,
      barberResponse: json['barberResponse'],
      barberRespondedAt: json['barberRespondedAt'] != null ? DateTime.parse(json['barberRespondedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'appointmentId': appointmentId,
      'barberId': barberId,
      'rating': rating,
      'comment': comment,
    };
  }
}
